audio/ffmpeg.py

```diff
@@ -0,0 +1,200 @@
+__all__ = ["BlissFFMPEGJob"]
+
+import copy
+import os
+import subprocess
+
+from sisyphus import *
+
+from recipe.i6_core.lib import corpus
+
+
+class BlissFFMPEGJob(Job):
+    """
+    Applies an FFMPEG audio filter to all recordigns of a bliss corpus.
+    This Job is extremely generic, as any valid audio option/filter string will work.
+
+    WARNING:
+        - This Job requires unique file names for the audio files
+        - Changing the duration of the audio files when you have multiple segments per audio,
+          as the segment information will be incorrect afterwards. Please use/write specific
+          jobs in that case (e.g. see corpus.data_augmentation.ChangeCorpusSpeed).
+
+    Typical applications:
+
+    **Changing Audio Format/Encoding**
+
+        - specify in `output_format` what container you want to use. If
+          the filter string is empty (""), ffmepg will automatically use a default encoding option
+
+        - specify specific encoding with `-c:a <codec>`. For a list of available codecs
+          and their options see https://ffmpeg.org/ffmpeg-codecs.html#Audio-Encoders
+
+        - specify a fixed bitrate with `-b:a <bit_rate>`, e.g. `64k`. Variable bitrate options depend on the
+          used encoder, refer to the online documentation in this case
+
+        - specify a sample rate with `-ar <sample_rate>`. FFMPEG will do proper resampling,
+          so the speed of the audio is NOT changed.
+
+
+    **Changing Channel Layout**
+
+        - for detailed informations see https://trac.ffmpeg.org/wiki/AudioChannelManipulation
+
+        - convert to mono `-ac 1`
+
+        - selecting a specific audio channel:
+          `-filter_complex \"[0:a]channelsplit=channel_layout=stereo:channels=FR[right]\" -map \"[right]\"`
+          For a list of channels/layouts use `ffmpeg -layouts`
+
+
+    **Simple Filter Syntax**
+
+    For a list of available filters see: https://ffmpeg.org/ffmpeg-filters.html
+
+    `-af \"<filter_name>=<first_param>=<first_param_value>:<second_param>=<second_param_value>\"`
+
+
+    **Complex Filter Syntax**
+
+    `-filter_complex \"[<input>]<simple_syntax>[<output>];[<input>]<simple_syntax>[<output>];...\"`
+
+    Inputs and outputs can be namend arbitrarily, but the default stream 0 audio can be accessed with [0:a]
+
+    The output stream that should be written into the audio is defined with `-map "[<output_stream>]"`
+
+    IMPORTANT! Do not forget to escape your quotation marks correctly for `-af` or `-filter_complex`
+
+    """
+
+    def __init__(
+        self,
+        corpus_file,
+        ffmpeg_option_string,
```

Comment:
I don't think this should be a string but rather a list of arguments (i.e. list of strings).

This would be much cleaner. And also it could contain maybe references to other files (for adding noise, or whatever FFmpeg can do).

Reply:
This is why wrote this should be a super-class for more specific jobs, which then deal with the parameters more explicitly. Adding noise is a bad example, because this would not work with this job in any case.

Reply:
I don't understand. In any case, this should not be a string, but a list of arguments (list of strings).